src/runtime_src/core/common/api/xrt_ip.cpp

```diff
@@ -286,6 +300,58 @@ class ip_impl
     return intr;
   }
 
+  void
+  add_callback(callback_function_type&& fcn)
+  {
+    if (callback)
+      throw xrt_core::error(-EINVAL, "xrt::ip::add_callback: Only one callback is allowed");
+
+    //Ensure interrupt object is created
+    auto intr = get_interrupt();
+    //Workaround needed
```

Comment:
Need more details on the work-around.   Like why.  Like when it will be fixed if ever.

Reply:
updated